website/docs/r/host_virtual_switch.html.markdown

```diff
@@ -0,0 +1,173 @@
+---
+layout: "vsphere"
+page_title: "VMware vSphere: vsphere_host_virtual_switch"
+sidebar_current: "docs-vsphere-resource-host-virtual-switch"
+description: |-
+  Provides a vSphere Host Virtual Switch Resource. This can be used to configure vSwitches direct on an ESXi host.
+---
+
+# vsphere\_host\_virtual\_switch
+
+The `vsphere_host_virtual_switch` resource can be used to manage vSphere
+standard switches on an ESXi host. These switches can be used as a backing for
+standard port groups, which can be managed by the
+[`vsphere_host_port_group`][host-port-group] resource.
+
+For an overview on vSphere networking concepts, see [this
+page][ref-vsphere-net-concepts].
+
+[host-port-group]: /docs/providers/vsphere/r/host_port_group.html
+[ref-vsphere-net-concepts]: https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.vsphere.networking.doc/GUID-2B11DBB8-CB3C-4AFF-8885-EFEA0FC562F4.html
+
+## Example Usages
+
+**Create a virtual switch with one active and one standby NIC:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics  = ["vmnic0"]
+  standby_nics = ["vmnic1"]
+}
+```
+
+**Create a virtual switch with extra networking policy options:**
+
+```hcl
+data "vsphere_datacenter" "datacenter" {
+  name = "dc1"
+}
+
+data "vsphere_host" "host" {
+  name          = "esxi1"
+  datacenter_id = "${data.vsphere_datacenter.datacenter.id}"
+}
+
+resource "vsphere_host_virtual_switch" "switch" {
+  name           = "vSwitchTerraformTest"
+  host_system_id = "${data.vsphere_host.host.id}"
+
+  network_adapters = ["vmnic0", "vmnic1"]
+
+  active_nics    = ["vmnic0"]
+  standby_nics   = ["vmnic1"]
+  teaming_policy = "failover_explicit"
+
+  allow_promiscuous      = false
+  allow_forged_transmits = false
+  allow_mac_changes      = false
+
+  shaping_enabled           = true
+  shaping_average_bandwidth = 50000000
+  shaping_peak_bandwidth    = 100000000
+  shaping_burst_size        = 1000000000
```

Comment:
🤔 not a blocker - we've got this in OPC land too, we probably want to expose an interpolation here to make it possible to specify this in MB and translate that to bytes?

Reply:
Yeah, might be something to take offline and think about, but yeah right now the API expects bytes, so that's what needs to be put in :stuck_out_tongue: